src/components/App.js

```diff
@@ -2,14 +2,32 @@ import React from 'react';
 import './App.css';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import Calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <div className="App">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  HandleClick(buttonName) {
+    const data = Calculate(this.state, buttonName);
+    this.setState(data);
+  }
+
+  render() {
+    const { total, next } = this.state;
+    return (
+      <div className="App">
+        <Display result={total || next} />
+        <ButtonPanel onClick={this.handleClick} />
```

Comment:
Your `handleClick` is declared with the uppercase `H` on line 17 of this file so make sure to make the function downcase.
That's the reason for the error every time you click any of the buttons of the calculator.